torch_np/testing/utils.py

```diff
@@ -207,924 +124,30 @@ def assert_equal(actual, desired, err_msg="", verbose=True):
-     y: array([1.        , 2.33333334])
-
-    """
-    __tracebackhide__ = True  # Hide traceback for py.test
-    from torch_np import imag, iscomplexobj, ndarray, real
-
-    # Handle complex numbers: separate into real/imag to handle
-    # nan/inf/negative zero correctly
-    # XXX: catch ValueError for subclasses of ndarray where iscomplex fail
-    try:
-        usecomplex = iscomplexobj(actual) or iscomplexobj(desired)
-    except ValueError:
-        usecomplex = False
-
-    def _build_err_msg():
-        header = "Arrays are not almost equal to %d decimals" % decimal
-        return build_err_msg([actual, desired], err_msg, verbose=verbose, header=header)
-
-    if usecomplex:
-        if iscomplexobj(actual):
-            actualr = real(actual)
-            actuali = imag(actual)
-        else:
-            actualr = actual
-            actuali = 0
-        if iscomplexobj(desired):
-            desiredr = real(desired)
-            desiredi = imag(desired)
-        else:
-            desiredr = desired
-            desiredi = 0
-        try:
-            assert_almost_equal(actualr, desiredr, decimal=decimal)
-            assert_almost_equal(actuali, desiredi, decimal=decimal)
-        except AssertionError:
-            raise AssertionError(_build_err_msg())
-
-    if isinstance(actual, (ndarray, tuple, list)) or isinstance(
-        desired, (ndarray, tuple, list)
-    ):
-        return assert_array_almost_equal(actual, desired, decimal, err_msg)
-    try:
-        # If one of desired/actual is not finite, handle it specially here:
-        # check that both are nan if any is a nan, and test for equality
-        # otherwise
-        if not (gisfinite(desired) and gisfinite(actual)):
-            if gisnan(desired) or gisnan(actual):
-                if not (gisnan(desired) and gisnan(actual)):
-                    raise AssertionError(_build_err_msg())
-            else:
-                if not desired == actual:
-                    raise AssertionError(_build_err_msg())
-            return
-    except (NotImplementedError, TypeError):
-        pass
-    if abs(desired - actual) >= np.float64(1.5 * 10.0 ** (-decimal)):
-        raise AssertionError(_build_err_msg())
-
-
-def assert_approx_equal(actual, desired, significant=7, err_msg="", verbose=True):
-    """
-    Raises an AssertionError if two items are not equal up to significant
-    digits.
 
-    .. note:: It is recommended to use one of `assert_allclose`,
-              `assert_array_almost_equal_nulp` or `assert_array_max_ulp`
-              instead of this function for more consistent floating point
-              comparisons.
+    from torch_np._normalizations import normalize_array_like
 
-    Given two numbers, check that they are approximately equal.
-    Approximately equal is defined as the number of significant digits
-    that agree.
+    actual = normalize_array_like(actual)
+    desired = normalize_array_like(desired)
+    return (actual == desired).all()
```

Comment:
Is it expected that `assert_*` functions don't actually assert anything but just return a bool? That seems a bit weird to me.

Reply:
Also, is it alright to return false for nans?